packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -119,10 +128,64 @@ export class SerializedStateManager {
 		} else {
 			const { baseSnapshot, snapshotBlobs } = this.pendingLocalState;
 			this.snapshot = { baseSnapshot, snapshotBlobs };
+			this.refreshSnapshot ??= (async () => {
+				this.latestSnapshot = await getLatestSnapshotInfo(
+					this.mc,
+					this.storageAdapter,
+					supportGetSnapshotApi,
+				);
+				this.updateSnapshotAndProcessedOpsMaybe();
+			})();
+
 			return { baseSnapshot, version: undefined };
 		}
 	}
 
+	/**
+	 * Updates class snapshot and processedOps if we have a new snapshot and it's among processedOps range.
+	 */
+	private updateSnapshotAndProcessedOpsMaybe() {
+		if (this.latestSnapshot === undefined) {
+			return;
+		}
+		this.snapshotRefreshed?.();
+		const snapshotSequenceNumber = this.latestSnapshot?.snapshotSequenceNumber;
+		if (this.processedOps.length === 0) {
+			// case in which we don't have to do nothing to processedOps
+			// since we don't have any.
+			this.snapshot = this.latestSnapshot;
```

Comment:
this seems wrong. its possible there are no op in process ops, but there are still ops that must be process between the current snapshot and the downloaded snapshot. it should be possible to write a test to simulate this. i think the right behavior is to do nothing.

Reply:
The scenario is, client downloads snapshot and not ops are processed. the client goes offline, and creates changes. the pending state is retrieved. the saved ops would be empty. when the pending state is loaded processed would be empty, but there could still be ops created remotely between the current snapshot, and the new snapshot